ml/src/main/scala/ai/h2o/sparkling/ml/algos/classification/H2OClassifier.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.h2o.sparkling.ml.algos.classification
+
+import ai.h2o.sparkling.H2OFrame
+import ai.h2o.sparkling.ml.algos.H2OAlgoCommonUtils
+import org.apache.spark.sql.Dataset
+import org.apache.spark.sql.functions.col
+import org.apache.spark.sql.types.StringType
+
+trait H2OClassifier extends H2OAlgoCommonUtils {
+  def getLabelCol(): String
+
+  override private[sparkling] def prepareDatasetForFitting(dataset: Dataset[_]): (H2OFrame, Option[H2OFrame]) = {
+    val labelColumnName = getLabelCol()
+    val preparedDataset = dataset.withColumn(labelColumnName, col(labelColumnName).cast(StringType))
```

Comment:
If I'm not wrong, H2O performs classification on if the label column is categorical. But since our conversion automatically converts string to categoricals, we should be ok, is that correct?

Reply:
Yes, exactly. 95% of string columns will be converted to categoricals. If string column contains only unique values, the conversion to categorical won't happen and model training will most likely fail. I was wondering whether I shouldn't convert label  column on H2OFrame explicitly, but H2O parsers won't convert every string column to categorical either. So IMHO, this approach mimics behavior of H2O-3.

Reply:
Yup, like it that way so it is not confusing to H2O users, thanks for explanation!